src/text-input/src/TextInput.js

```diff
@@ -118,55 +106,28 @@ class TextInput extends PureComponent {
     const borderRadius = theme.getBorderRadiusForControlHeight(height)
 
     return (
-      <React.Fragment>
-        <Text
-          is="input"
-          className={cx(themedClassName, className)}
-          type="text"
-          size={textSize}
-          width={width}
-          height={height}
-          required={required}
-          disabled={disabled}
-          placeholder={placeholder}
-          value={this.getValue()}
-          {...(onChange
-            ? { onChange }
-            : { onChange: e => this.setState({ value: e.target.value }) })}
-          paddingLeft={Math.round(height / 3.2)}
-          paddingRight={Math.round(height / 3.2)}
-          borderRadius={borderRadius}
-          spellCheck={spellCheck}
-          aria-invalid={isInvalid}
-          {...(disabled ? { color: 'muted' } : {})}
-          css={css}
-          {...props}
-        />
-        {isClearable &&
-          (!this.props.value &&
-            this.state.value.length > 0 &&
-            !this.props.onChange) && (
-            <Icon
-              color="muted"
-              icon="cross"
-              appearance="default"
-              onClick={() => {
-                this.setState({ value: '' })
-              }}
-              marginLeft={-20}
-            />
-          )}
-        {isClearable &&
-          (this.props.value && this.props.onChange) && (
-            <Icon
-              color="muted"
-              icon="cross"
-              appearance="default"
-              onClick={onClear}
-              marginLeft={-20}
-            />
-          )}
-      </React.Fragment>
+      <StatelessTextInput
+        value={this.state.value}
+        isClearable={isClearable}
+        onClear={() => this.setState({ value: '' })}
+        onChange={e => this.setState({ value: e.target.value })}
+        className={cx(themedClassName, className)}
```

Comment:
I am all for consolidating these props into the ...restProps type reference, but again was not positive on the usage of the Stateless version and how this might look to consumers. 

I often inspect a render method when I'm looking at a library and while the spread might be cleaner it is less explicit that this component I'm consuming does XYZ with all these props. 

Happy to clean this up though.